src/mem3/src/mem3_rpc.erl

```diff
@@ -378,20 +378,34 @@ rexi_call(Node, MFA, Timeout) ->
     Mon = rexi_monitor:start([rexi_utils:server_pid(Node)]),
     Ref = rexi:cast(Node, self(), MFA, [sync]),
     try
-        receive
-            {Ref, {ok, Reply}} ->
-                Reply;
-            {Ref, Error} ->
-                erlang:error(Error);
-            {rexi_DOWN, Mon, _, Reason} ->
-                erlang:error({rexi_DOWN, {Node, Reason}})
-        after Timeout ->
-            erlang:error(timeout)
-        end
+        wait_message(Node, Ref, Mon, Timeout)
     after
         rexi_monitor:stop(Mon)
     end.
 
+wait_message(Node, Ref, Mon, Timeout) ->
+    receive
+        Msg ->
+            process_raw_message(Msg, Node, Ref, Mon, Timeout)
+    after Timeout ->
+        erlang:error(timeout)
+    end.
+
+process_raw_message(Msg0, Node, Ref, Mon, Timeout) ->
+    {Msg, Delta} = csrt:extract_delta(Msg0),
+    csrt:accumulate_delta(Delta),
+    case Msg of
+        {Ref, {ok, Reply}} ->
+            Reply;
+        {Ref, Error} ->
+            erlang:error(Error);
+        {rexi_DOWN, Mon, _, Reason} ->
+            erlang:error({rexi_DOWN, {Node, Reason}});
+        Other ->
+            ?LOG_UNEXPECTED_MSG(Other),
+            wait_message(Node, Ref, Mon, Timeout)
```

Comment:
Good idea to log the message if we drop it.

Hmm, maybe we should see if we can use aliases for this to avoid un-expected messages. If we're switching to dropping messages we should make it a separate (preliminary) PR to discuss it and then rebase the work on that.